olca-app/src/org/openlca/app/devtools/csharp/server/ClientSender.java

```diff
@@ -0,0 +1,88 @@
+package org.openlca.app.devtools.csharp.server;
+
+
+/**
+ * Sends messages to the client. Messages are stored in a message queue. When
+ * the queue is empty, ClientSender falls in sleep until a new message is
+ * arrived in the queue. When the queue is not empty, ClientSender sends the
+ * messages from the queue to the client socket.
+ */
+
+import java.io.*;
+import java.net.*;
+import java.util.*;
+
+public class ClientSender extends Thread {
+	private Vector mMessageQueue = new Vector();
```

Comment:
maybe directly using a queue would be nice, e.g. https://docs.oracle.com/javase/7/docs/api/java/util/ArrayDeque.html -> implements Queue